heaps/min_heap.py

```diff
@@ -15,25 +17,38 @@ class MinHeap:
     def __init__(self):
         self.store = []
 
-
     def add(self, key, value = None):
-        """ This method adds a HeapNode instance to the heap
-            If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+        """ Adds a HeapNode instance to the heap
+            If value == None the new node's value is set to key
+            Time Complexity: O(log n)
+            Space Complexity: O(n)
         """
-        pass
+        if value == None: 
+            value = key
+
+        node = HeapNode(key, value)
+
+        self.store.append(node)
+
+        self.heap_up(len(self.store) - 1)
 
     def remove(self):
```

Comment:
✨ However space complexity will be O(log n) because of the recursive call stack of `heap_down`